plane/src/database/connect.rs

```diff
@@ -304,6 +310,12 @@ async fn attempt_connect(
         .or(default_cluster)
         .ok_or(ConnectError::NoClusterProvided)?;
 
+    if let Some(subdomain) = &spawn_config.subdomain {
+        if subdomain.contains('.') {
+            return Err(ConnectError::InvalidSubdomain);
+        }
```

Comment:
Subdomains can consist of `a-z`, `0-9`, and `-` but not as the first or last char. How about this:

```rust
let valid_subdomain = s.chars().all(|c| c.is_alphanumeric() || c == '-')
    && !s.starts_with('-') && !s.ends_with('-')
if !valid_subdomain {...}
```